pantheon-sessions.php

```diff
@@ -342,12 +342,186 @@ public static function check_native_primary_keys() {
+		if ( ! is_multisite() ) {
+			$this->safe_output( __( 'Single site detected. Beginning processing... \n', 'wp-native-php-sessions' ), 'log' );
+
+			$this->add_single_index( $wpdb->prefix );
+
+			$this->safe_output( __( 'Operation complete, please verify that your site is working as expected. When ready, run wp pantheon session primary-key-finalize to clean up old data, or run wp pantheon session primary-key-revert if there were issues.', 'wp-native-php-sessions' ), 'log' );
+
+			return;
+		}
+
+		$this->safe_output( __( 'Multisite installation detected. Processing Sites individually...', 'wp-native-php-sessions' ), 'log' );
+
+		$output = [
+			'no_session_table' => 0,
+			'id_column_exists' => 0,
+		];
+		$site_list = $this->get_all_sites( $start_position );
+		$site_count = count( $site_list );
+
+		for ( $i = $start_position; $i < $site_count; $i++ ) {
+			// translators: %s is the current blog, and then the array index of the current site.
+			$this->safe_output( __( 'Processing site %d. In the event of a timeout or error, resume execution starting from this point via "wp pantheon session add-index --start_point=%d". To skip this site if it does not need processing, run "wp pantheon session add-index --start_point=%d".', 'wp-native-php-sessions' ), 'log', [ $site_list[ $i ], $i, $i + 1 ] );
+
+			$blog_prefix = $wpdb->get_blog_prefix( $site_list[ $i ] );
+			$output = $this->add_single_index( $blog_prefix, $output, true );
+
+			// translators: %s is the current site id.
+			$this->safe_output( __( 'Processing for site %d complete.', 'wp-native-php-sessions' ), 'log', [ $site_list[ $i ] ] );
+		}
+
+		if ( $output['no_session_table'] != 0 ) {
+			// translators: %d is the number of sites which did not have a session table.
+			$this->safe_output( __( '%d site(s) did not have a session table and were skipped. If this was unexpected, please review output.', 'wp-native-php-sessions' ), 'log', [ $output['no_session_table'] ] );
+		}
+
+		if ( $output['id_column_exists'] != 0 ) {
+			// translators: %d is the number of sites which had an ID column already.
+			$this->safe_output( __( '%d site(s) already had an ID column and were skipped. If this was unexpected, please review output.', 'wp-native-php-sessions' ), 'log', [ $output['id_column_exists'] ] );
+		}
+
+		$this->safe_output( __( 'Operation complete, please verify that your site is working as expected. When ready, run wp pantheon session primary-key-finalize to clean up old data, or run wp pantheon session primary-key-revert if there were issues.', 'wp-native-php-sessions' ), 'log' );
+	}
+
+	/**
+	 * Retrieves all sites with query batches.
+	 *
+	 * @param int $start_position Starting position of query.
+	 *
+	 * @return array
+	 */
+	public function get_all_sites( $start_position ) {
+		$num_sites = 2;
+		$sites = ( $start_position == 0 ) ? [] : array_fill( 0, $start_position, null );
+		$page = 0;
+
+		while ( true ) {
+			$offset         = ( $page * $num_sites ) + $start_position;
```

Comment:
```suggestion
			$offset = ( $page * $num_sites ) + $start_position;
```
I suppose this is an artifact of spacing linting?

Reply:
It was, but it also doesn't seem to mind if I put it down to one space and it looks better like that.